File: Examples/VisualBasic/Images/BuildDocAndInsertImage.vb
Imports System.IO
Imports  Aspose.Note
Imports System.Collections.Generic
Imports System.Drawing
Namespace Images
    Public Class BuildDocAndInsertImage
        Public Shared Sub Run()
            ' ExStart:BuildDocAndInsertImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Images()

            ' Create an object of the Document class
            Dim doc As New Document()
            ' Initialize Page class object
            Dim page As New Page(doc)
            ' Initialize Outline class object and set offset properties
            Dim outline As New Outline(doc) With {
                 .VerticalOffset = 0,
                 .HorizontalOffset = 0
            }
            ' Initialize OutlineElement class object
            Dim outlineElem As New OutlineElement(doc)
            ' Load an image by the file path.
            Dim image As New Image(doc, dataDir & Convert.ToString("image.jpg"))
            ' Set image alignment
            image.Alignment = HorizontalAlignment.Right
            ' Add image
            outlineElem.AppendChild(image)
            ' Add outline elements
            Outline.AppendChild(outlineElem)
            ' Add Outline node
            page.AppendChild(Outline)
            ' Add Page node
            doc.AppendChild(page)

            dataDir = dataDir & Convert.ToString("BuildDocAndInsertImage_out_.one")
            ' Save OneNote document
            doc.Save(dataDir)
            ' ExEnd:BuildDocAndInsertImage 
            Console.WriteLine(Convert.ToString(vbLf & "Document created and image inserted successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Images/BuildDocAndInsertImage.cs
using System;

namespace Aspose.Note.Examples.CSharp.Images
{
    public class BuildDocAndInsertImage
    {
        public static void Run()
        {
            // ExStart:BuildDocAndInsertImage
            // ExFor:Document
            // ExFor:Image
            // ExFor:Image.Alignment
            // ExSummary:Shows how to add an image from file to a document.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Images();

            // Create an object of the Document class
            Document doc = new Document();

            // Initialize Page class object
            Page page = new Page();

            // Initialize Outline class object and set offset properties
            Outline outline = new Outline();

            // Initialize OutlineElement class object
            OutlineElement outlineElem = new OutlineElement();
            
            // Load an image by the file path.
            Image image = new Image(dataDir + "image.jpg")
                                      {
                                          // Set image alignment
                                          Alignment = HorizontalAlignment.Right
                                      };

            // Add image
            outlineElem.AppendChildLast(image);
            
            // Add outline elements
            outline.AppendChildLast(outlineElem);
            
            // Add Outline node
            page.AppendChildLast(outline);
            
            // Add Page node
            doc.AppendChildLast(page);

            // Save OneNote document
            dataDir = dataDir + "BuildDocAndInsertImage_out.one";
            doc.Save(dataDir);
            
            // ExEnd:BuildDocAndInsertImage 
            
            Console.WriteLine("\nDocument created and image inserted successfully.\nFile saved at " + dataDir); 
        }
    }
}